### DISCUSSION TITLE
Time / Day displaying as appointment

### DISCUSSION THREAD

**Discussion by whateverx28** [code4recovery/12-step-meeting-list#1355] [Q&A] [answered]
Hello, I imported a .csv file into the 12 Step Meeting List plugin but neither the date nor the time show up. The column only says appointment. I checked the csv and the Date and Time are in the right columns and formatted properly. Maybe I left too many columns in it? Please also tell me how to delete the csv I imported because it's not working. I don't see how to delete it. Thank youl
[RIC VA AA Mtgs.csv](https://github.com/code4recovery/12-step-meeting-list/files/14302462/RIC.VA.AA.Mtgs.csv)

**Comment by whateverx28** [comment]
I found your directions on the FAQ page, and exported the file to downloads. Then I deleted it manually, though it took a long time. I think there must be a quicker way to do this. I reviewed the csv file in which I cleaned the data and formatted it according to the 12 Step template. Yet neither the date nor the time showed up only the words Appointment when I imported it. Thanks for your comments and help. Karen

**Reply by tim-rohrer** [reply]
Please note on the import page that you have options for how the process should handle existing data. The default should overwrite all existing meeting data.

**Comment by kiyote33** [comment]
Hi Karen,
I've reformatted the first 117 records from your csv into another file here and it may provide some help for you. Note that I concatenated your street, city, state, zip, country info into the single address column. It shows correctly on my test site. Hope this helps...
[meetings(Richmond).csv](https://github.com/code4recovery/12-step-meeting-list/files/14304027/meetings.Richmond.csv)

**Comment by whateverx28** [comment]
Wayne,

Thank you! There are about 800 records/meetings in this area, however, and I shold concatenated the rest of them. I concatenated the time and am/pm before I imported the first time into the plugin, but that was relatively easy. 

Can you provide me the formula to concatenate the remaining records/meetings? Example, I also have no idea how to add USA, etc.

I also moved all the zoom meetings in the address section in the ss I worked on before I received the one you did. Is that a good idea? 

Shouldn’t I also remove all the columns K through AQ before I try to import, that is, after I concatenate the remaining records, again?

Thank you very much! 

Karen

**Reply by kiyote33** [reply]
Hi Karen @whateverx28 , see my detailed response in our [discussion forum](https://github.com/code4recovery/12-step-meeting-list/discussions/1355). Yes, you could remove all the columns after the Notes field. The importer code will recreate many of the fields you see there. 

The emails you receive from us are auto-generated by Github which are difficult for us to track. Our team here at Code For Recovery can more easily track communications and provide help when you reply back in the Discussion Forum channel. Thx...

**Comment by whateverx28** [comment]
Tim,

Thank you for this info. I manually deleted the data I earlier imported, so there should be nothing left in the db.

Karen

**Comment by whateverx28** [comment]
PS Does it matter at all that all the data in the column Region is formatted uppercase? I know how to make it all proper case, and I did that with most of the data but not with the Region data. When I exported the data from the current old site built in 2007, it all came in uppercase.

**Reply by kiyote33** [reply]
I prefer proper case too. All upper casing (SHOUTING) is a little rude but that's the fun of website design in recovery, figuring out what works and what doesn't for your group conscience.

**Comment by whateverx28** [comment]
You guys are so kind and are helping Richmond AA so much! I am going to donate and ask our web committee to donate. I will tell them about you!

Karen

**Comment by kiyote33** [comment]
Hi @whateverx28, I just took a break for dinner. The reformatting process I used was this:

1. Renamed LocationCity to Region. Duplicated Region and renamed the duped column to Sub Region.
2. Renamed Address column to Street.
3. Replaced all the Southside, Northside, West End, and East End in Regions column with RICHMOND.
4. Removed all the cities from Sub Region, leaving only the 4 sub regions occurrences and empty cells.
5. Insert a new column called State and populate the entire column with the state initials: VA.
6. Insert a new column called Country and populate the entire column with the country initials: USA.
7. Duplicated Region column into a new column named City. 
8. Move columns so I now have 5 columns: Street, City, State, Zip, Country in columns F through J.
9. Insert a new column called Address and populate the 2nd row cell with a concatenation formula: 
**=concatenate(F2,", ",G2,", ",H2,", ",I2,", ",J2)**. Select from the 2nd cell to the last cell in the column and hit the CNTL+D keys to populate all the cells with the concatenation formula.

The problem with the day and time columns required two new columns be created. The first one had the cells formatted for time with a cell format set to 1:00 AM and the second column cell format was set to Text. I copied the Time column from your spreadsheet into the first new column that I named Time, and all the column data from your spreadsheet Day column into the second new column that I likewise named Day.
I renamed the Meeting column to Name.
I then ran the new file through the Importer with the last delete option to "delete all meetings, locations, groups, districts, and regions".
The last thing I did was to Export the database, which is the file you received up above here.
Hope this helps. I know there was a lot there.

**Reply by kiyote33** [reply]
PS... I'm not at all familiar with your area in Virginia, other than Richmond being a city. Many of the names I thought were regions may well be sub regions of Richmond or another city.

**Reply by whateverx28** [reply]
Thank you, thank you! I'm working on it now. You guys are the best!

**Reply by whateverx28** [reply]
Kiyote33, Thank you a million for these directions. I have three questions: 

1. The day and time columns are formatted exactly as you indicate, so do I have to make new ones? 
2. I don't have LocationCity so I don't know what to rename to Region. I have Name (which you suggested) which tells user the name of the group, Location which specifies name of church or building, Address which specifies exact address, etc. This means I don't know how to follow your directions in Item 1.

I can follow your directions in items 3 thru 9, so that's good. Thanks again, Karen

**Reply by kiyote33** [reply]
@whateverx28, I was not able to use the Time and Day columns from your CSV. I tried to no avail but something unbeknown  to me has corrupted them.  Out of desperation, I created two new columns, formatted the cells accordingly,  and copy/pasted the data from the old columns into the new ones. If you can figure out what is corrupting the old columns, you may not need to create new ones. I wasn't able to do that.
Here is the header and top of the CSV I was working from:
![image](https://github.com/code4recovery/12-step-meeting-list/assets/57125427/0b9f022e-63be-419f-9468-3b6dc700bd32)
Yes, 'Meeting' should be 'Name', 'Location / Group' can be set to just 'Location', and 'LocationCity' would be simply 'Region'. You could insert an empty column next to the renamed 'Region' column and simply move the four quadrant locations "SOUTHSIDE, NORTHSIDE, WEST END, AND EAST END" to that column after naming it your Sub Region column.  Note that all Sub Regions must have a parent Region.

Hope this helps. Let me know if anything else is unclear. Good Luck!

**Reply by whateverx28** [reply]
kiyote33 -- I get it about the corruption. Thank you for identifying the problem. I'll go with your idea to create new and copy/paste. Thank you for this great help. Karen

**Comment by joshreisner** [comment, marked as answer]
sorry to add another voice here, but i observed the same issue and i'm not exactly sure why the meetings where showing up as appointment.

i resolved it by creating a [google sheet](https://docs.google.com/spreadsheets/d/1Cf81EPLy3Ii7GtidaLw0MmPyvVi2rl-pCLuRpg3dtSk/edit#gid=2048875353) from the original CSV you uploaded @whateverx28 and then formatting the time column with Format > Number > 3:59 PM (see screenshot below)

i also made a few of the column edits @kiyote33 suggested above. i also added a `Conference URL` column (online meeting urls can't go in the address column)

i then saved it as CSV and was able to import it with no Appointment meetings: [Richmond - RIC.VA.AA.Mtgs(1).csv](https://github.com/code4recovery/12-step-meeting-list/files/14314506/Richmond.-.RIC.VA.AA.Mtgs.1.csv)

there are still some issues in the data however with online meetings.  you'll want to make sure all the links start with `https://` and that they all have at least a city and a location name even if that's "Richmond" and "Online Meeting" respectively

hope it helps

<img width="682" alt="Screenshot 2024-02-16 at 10 00 07 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/ae7cec3a-4501-4266-b97a-40d852039d1d">

**Reply by joshreisner** [reply]
you're welcome to edit the Google Sheet directly and re-export to CSV, i set it so anyone with the link can edit

**Reply by whateverx28** [reply]
Josh, helps a lot what you did! Looks like I'll go with Google sheets now instead of MS Excel. Questions and comments below:

1. I'll reformat online mtgs with https://, but where should I indicate "Richmond" or "Online Meeting"? Do I make another column to do that? 
3. Is there a way to automate adding the https:// because there are quite a few of them. I added https:// to two of them but they become clickable immediately. I can right click and remove the link but keep the formatting. Is that a good thing to do, or does it matter? Leave it clickable?
4. Kiyote33 showed me how to add USA and concatenate with address. I will do this, too.

Thank you very much!

**Reply by whateverx28** [reply]
Joshreisner, one more question. The Google ss you sent opens up immediately to time 6:00 PM on Thursday (see attached screenshot). I don't see a principle of organization. I know how to reorder ascending/descending but I am afraid of wrecking the great work you did. Maybe there is a reason it opens up to 6:00 PM on Thursday. Maybe it doesn't matter when imported. Thank you!
![Google ss screenshot](https://github.com/code4recovery/12-step-meeting-list/assets/158858278/7a3c4d90-37e9-4adc-842f-03c628b2ea74)

**Reply by joshreisner** [reply]
hi @whateverx28 ok great! to answer your questions:
* i sorted the document a few times, feel free to do that
* no need to concatenate addresses, they can stay how they are
* put Richmond in the City column
* put "Online Meeting" in the Location column if that's empty
* Conference URL: if the goal is to import this into TSML, then don't change these column names. your users will not see this sheet, they will see the meeting finder on your website
* i don't know of an easy way to prepend `https://` on the fields that are missing that value, probably quickest just to sort the sheet by Conference URL and go down the list

**Reply by whateverx28** [reply]
joshreisner, I think you mean make a separate column for Online Meeting where I enter "Richmond" or "Online Meeting", so it works with the zoom meeting URLs in the Conference URL column.. I hope I am not confusing the issue.

**Reply by joshreisner** [reply]
no need to create any new columns! happy to do a zoom with you if it would be helpful so we can collab in real time

if it's an online meeting with no street address i would make the
* City => Richmond
* State => VA
* Location => Online Meeting
* Address => (empty)
* Conference URL => (something that starts with `https://`)

**Reply by whateverx28** [reply]
A zoom would be very helpful. What looks good to you?

**Reply by joshreisner** [reply]
i could meet at 4pm ET / 1pm PT or later today, or starting tomorrow at 12pm ET / 9am PT?

**Reply by whateverx28** [reply]
4pm ET today is perfect. I cannot do tomorrow at 12pm ET.

**Reply by whateverx28** [reply]
I have a free zoom account if you want to use that.

**Reply by joshreisner** [reply]
ok great. i'll drop a zoom link on here in ~20 mins

**Reply by whateverx28** [reply]
👍

**Comment by whateverx28** [comment]
joshreisner -- OK to change column name from "Conference URL" to "Online Meetings"? A newbie not quite sober will recognize "Online Meetings" but maybe not as easily "Conference URL". Thanks for your help.

**Comment by whateverx28** [comment]
johreisner -- please send me the final spreadsheet (I think Google and not MS). Thank you a million!!

**Reply by joshreisner** [reply]
it's the same sheet linked above: https://docs.google.com/spreadsheets/d/1Cf81EPLy3Ii7GtidaLw0MmPyvVi2rl-pCLuRpg3dtSk/edit#gid=2048875353

**Reply by whateverx28** [reply]
Thank you!

**Comment by whateverx28** [comment]
Josh,

Regarding this meeting list < https://aarichmond2stg.wpenginepowered.com/meetings/> for Richmond VA, there is a problem I think it’s critical for our members (especially new ones) to fix. Website is using Version 3.14.33 of the 12 Step Meeting List with TSML UI enabled and Mapbox.

The current website <aarichmond.org> lists 48 areas within five districts. It’s clearly not viable to include all 48 in the 12-Step plugin. I am stuck with what the current webmaster did in terms of identifying areas, but these five below cover the general geographical area called Richmond VA. 

The current 12 Step Meeting list created with the plugin and your help lists only two “areas,” Richmond, and Manakin-Sabot.  I think it’s important, however, to list at least five areas under the column “Anywhere” in the meeting list. The remaining I want to include are Glen Allen, Southside, Mechanicsville, and Ashland. Richmond is already included, and Manakin-Sabot should be removed.

In the backend of this plugin, I see three regions, Henrico, Richmond, and Manakin-Sabot, but no districts. My fault entirely but I have to correct this. I think the only “region” is Richmond, but Henrico and Manakin-Sabot are not regions and not even districts and should be removed. I think all five districts should be listed, but I don’t know what happens to the plugin meeting list if I list them, since I am not going to include 48 areas.

I don’t know how to fix this and am asking your help. I have the meeting list as a Google ss but will have to rework it to filter for the above areas, and am not sure I can do this well, let alone reload it into the plugin. Thank you for taking a look at it. 

Karen

**Reply by joshreisner** [reply]
Ok! i would create a new column called Region in your google sheet and start by copying the values from the City column, then adjust them as necessary, and re-import

**Comment by whateverx28** [comment]
joshreisner, 

Thank you very much for responding. I hope you have time to answer these
questions. I reviewed the support forums to find any issues like mine but
don't see anything that helps. 

1.      Should I insert a column "Region" before these columns | City |
State | Country | Postal Code |, between these columns, or after these four
columns? 

2.      Region will hold these five regions, Ashland, Glen Allen,
Mechanicsville, Richmond, Southside, except that Richmond is also the only
value in the City column .

3.      You comment, "copy the values from the City column, then adjust as
necessary." There is only one value in the City column which is Richmond. 

4.      Does it matter if the regions and/or districts are accurately
entered in the backend under Settings?

**Reply by joshreisner** [reply]
hi karen,

it doesn't matter where you create the Region column, and it doeasn't matter how you enter that information in there, whatever is easiest for you!

the importer should create any missing regions for you

j
